java/src/jmri/jmrit/beantable/SignalMastLogicTableAction.java

```diff
@@ -251,37 +231,36 @@ public int getColumnCount() {
 
             @Override
             public void setValueAt(Object value, int row, int col) {
-                if (col == COMCOL) {
-                    SignalMastLogic rowLogic = getLogicFromRow(row);
-                    assert rowLogic != null;
-                    rowLogic.setComment((String) value, getDestMastFromRow(row));
-                } else if (col == EDITLOGICCOL) {
-                    class WindowMaker implements Runnable {
-
-                        final int row;
-
-                        WindowMaker(int r) {
-                            row = r;
-                        }
-
-                        @Override
-                        public void run() {
+                SignalMastLogic rowLogic = getLogicFromRow(row);
+                if ( rowLogic == null ){
+                    return;
+                }
```

Comment:
PR looks good. My only question is if this could break things. If I read the code correctly, the check for if `rowLogic` is null was not done for every column, but now the check is done for every column. Could this cause a problem?

I don't know this code and haven't had time to check, so I ask only to make sure nothing breaks.

Reply:
As always thanks for looking over @danielb987 .
This is the Signal Mast Logic Table, so if the Signal Mast Logic isn't available for a row ( in process of being deleted? ), then no problem with returning without actioning anything.